packages/cheqd-blockchain-modules/src/attest/internal.js

```diff
@@ -24,29 +24,15 @@ export default class CheqdInternalAttestModule extends createInternalCheqdModule
   };
 
   async attest(did, attestId) {
-    const stringDid = CheqdDid.from(did).toEncodedString();
-    const stringAttestId = String(TypedUUID.from(attestId));
-    const item = await this.query.resource(stringDid, stringAttestId);
-
-    return option(Iri).from(item?.resource?.data);
+    return option(Iri).from(
+      (await this.resource(did, attestId))?.resource?.data,
+    );
   }
 
   async attestId(did) {
-    let resources = [];
-    let paginationKey;
-
-    do {
-      // eslint-disable-next-line no-await-in-loop
-      const res = await this.query.collectionResources(
-        CheqdDid.from(did).toEncodedString(),
-        paginationKey,
-      );
-      resources = resources.concat(
-        res.resources.filter((resource) => resource.resourceType === 'attest'),
-      );
-      ({ paginationKey } = res);
-    } while (!resources.length && paginationKey != null);
-
-    return resources.find((resource) => !resource.nextVersionId)?.id;
+    return await this.latestResourceIdBy(
```

Comment:
return await isnt needed, can just return the promise

Reply:
Using `return await` ensures the stack trace is captured accurately, reflecting where the promise awaits, simplifying debugging.